### PR TITLE
AIPerception Team Sense blueprint exposed function

### PR DESCRIPTION
**Pull request by jagged-teeth**
AISense_Team BP exposed function to replicate senses to friendlies

- - -
This change is [Reviewable](https://reviewable.io/reviews/TheEmidee/UEAIExtensions/16)

### CODE REVIEW

File: Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿#pragma once
+
+#include "CoreMinimal.h"
+#include "Kismet/BlueprintFunctionLibrary.h"
+#include "UObject/ObjectMacros.h"
+
+#include "AIExtTeamEvent.generated.h"
+
+UCLASS()
+class AIEXTENSIONS_API UAIExtTeamEvent final : public UBlueprintFunctionLibrary
```
Comment (Frickerson) [inline Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h]: Should we create a new BP func library just for this?

File: Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿#include "AIPerception/AIExtTeamEvent.h"
+
+#include "Perception/AIPerceptionSystem.h"
```
Comment (TheEmidee) [inline Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp]: <>

File: Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+#include "Perception/AISense_Team.h"
```
Comment (TheEmidee) [inline Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp]: <>

File: Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+void UAIExtTeamEvent::ReportTeamEvent( UObject * WorldContextObject, AActor * InBroadcaster, AActor * InEnemy, const FVector & InLastKnowLocation, float EventRange, float PassedInfoAge, float InStrength )
```
Comment (TheEmidee) [inline Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp]: function arguments are snake_case

File: Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+{
+    UAIPerceptionSystem * PerceptionSystem = UAIPerceptionSystem::GetCurrent( WorldContextObject );
+    if ( PerceptionSystem )
```
Comment (TheEmidee) [inline Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp]:
```
if ( auto * perception_system = XXX )
{

}
```

File: Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    {
+        FAITeamStimulusEvent Event( InBroadcaster, InEnemy, InLastKnowLocation, EventRange, PassedInfoAge, InStrength );
```
Comment (TheEmidee) [inline Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp]: snake_case

File: Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp (hunk 4 above)
Comment (TheEmidee) [reply Source/AIExtensions/Private/AIPerception/AIExtTeamEvent.cpp]: also no prefix with `In` or `Out`

File: Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿#pragma once
+
+#include "CoreMinimal.h"
```
Comment (TheEmidee) [inline Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h]: <>

File: Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+#include "Kismet/BlueprintFunctionLibrary.h"
```
Comment (TheEmidee) [inline Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h]: <>

File: Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+#include "UObject/ObjectMacros.h"
```
Comment (TheEmidee) [inline Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h]: <>

File: Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h (hunk 1 above)
Comment (TheEmidee) [reply Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h]: what do you mean?

**Review by TheEmidee — CHANGES_REQUESTED** [review]

**Review by TheEmidee — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TheEmidee/UEAIExtensions/16)* status: 1 of 2 files reviewed, 2 unresolved discussions (waiting on @jagged-teeth and @VectorGamez)

___
*[`Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h` line 10 at r2](https://reviewable.io/reviews/TheEmidee/UEAIExtensions/16#-NzDgAfF16Q6xIAGUXlf:-NzDgAfF16Q6xIAGUXlg:b-oe949a) ([raw file](https://github.com/TheEmidee/UEAIExtensions/blob/a018a89bdd63c53a80a6f4c63ace6fd576ff21da/Source/AIExtensions/Public/AIPerception/AIExtTeamEvent.h#L10)):*
> ```objc
> 
> UCLASS()
> class AIEXTENSIONS_API UAIExtTeamEvent final : public UBlueprintFunctionLibrary
> ```

rename to UAIExtPerceptionFunctionLibrary

**Review by TheEmidee — COMMENTED** [review]
> Reviewed 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/TheEmidee/UEAIExtensions/16)* status: all files reviewed, 2 unresolved discussions (waiting on @jagged-teeth and @VectorGamez)

**Review by TheEmidee — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TheEmidee/UEAIExtensions/16#-:-NzE4tz9-q56UtOv5ZuG:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @VectorGamez)

**Pull request merged**